lib/controllers/AssetCategoryController.ts

```diff
@@ -105,7 +108,17 @@ export class AssetCategoryController extends RelationalController {
 
     const embeddedMetadata = this.getFieldPath(request, 'AssetCategory', '_metadataId', 'metadata');
     const assetCategoryContainer = this.getFieldPath(request, 'assetMetadatas');
-    return super.genericLink(request, embeddedMetadata, assetCategoryContainer, true);
+    const r = await super.genericLink(request, embeddedMetadata, assetCategoryContainer, true);
+    if (request.input.body && request.input.body.value) {
+      const updateRequest = {
+        metadatasValues: {}
+      };
+      updateRequest.metadatasValues[request.getString('_metadataId')] = request.input.body.value;
+      request.input.body = updateRequest;
+      await this.update(request);
+    }
+    return r;
```

Comment:
It's more readable to explicitly return the result of the API action so future developers know instantly what this API action is returning without having to look into `super.genericLink`

```
const { some, property } = await this.service.someMethod();

return { some, property };
```

Reply:
return not useful and removed